tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/DataConnections.resource

```diff
@@ -0,0 +1,74 @@
+${S3_SECRET_DC_INPUT_XP}=          xpath=//input[@aria-label="AWS field AWS_SECRET_ACCESS_KEY"]
+${S3_ENDPOINT_DC_INPUT_XP}=          xpath=//input[@aria-label="AWS field AWS_S3_ENDPOINT"]
+${S3_REGION_DC_INPUT_XP}=          xpath=//input[@aria-label="AWS field AWS_DEFAULT_REGION"]
+${DC_SECTION_XP}=             xpath=//div[div//h4[@id="data-connections"]]
+${DC_ADD_BTN_1_XP}=           ${DC_SECTION_XP}/div/button[text()="Add data connection"]
+${DC_ADD_BTN_2_XP}=           xpath=//footer/button[text()="Add data connection"]
+
+
+*** Keywords ***
+Create S3 Data Connection
+    [Documentation]    Creates a S3 Data Connection from DS Project details page
+    [Arguments]    ${project_title}    ${dc_name}    ${aws_access_key}    ${aws_secret_access}    
+    ...            ${aws_s3_endpoint}    ${aws_region}    ${connected_workbench}=${NONE}    ${press_cancel}=${FALSE}
+    Click Button    ${DC_ADD_BTN_1_XP} 
+    Wait Until Page Contains Element    ${S3_NAME_DC_INPUT_XP}
+    Input Text    ${S3_NAME_DC_INPUT_XP}    ${dc_name}
+    Input Text    ${S3_KEY_DC_INPUT_XP}    ${aws_access_key}
+    Input Text    ${S3_SECRET_DC_INPUT_XP}    ${aws_secret_access}
+    Input Text    ${S3_ENDPOINT_DC_INPUT_XP}    ${aws_s3_endpoint}
+    Input Text    ${S3_REGION_DC_INPUT_XP}    ${aws_region}
+    IF    ${press_cancel} == ${TRUE}
+        Click Button    ${GENERIC_CANCEL_BTN_XP}
+    ELSE
+        Wait Until Element Is Enabled    ${STORAGE_ADD_BTN_2_XP}
+        Click Button    ${STORAGE_ADD_BTN_2_XP}
+    END
+    Wait Until Modal Disappears
+    Wait Until Project Is Open    project_title=${project_title}
+
+Data Connection Should Be Listed
+    [Documentation]    Checks a Data Connection is listed in DS Project details page
+    [Arguments]     ${name}   ${type}   ${connected_workbench}
+    Run keyword And Continue On Failure     Page Should Contain Element    
+    ...    ${DC_SECTION_XP}//td[@data-label="Name"]/h4/div[text()="${name}"]
+    Run keyword And Continue On Failure     Page Should Contain Element
+    ...    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/h4/div[text()="${name}"]]/td[text()=" ${type}"]
+    IF    "${connected_workbench}" == "${NONE}"
+        Run Keyword And Continue On Failure    Page Should Contain Element    
+        ...    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/h4/div[text()="${name}"]]/td[text()="No connections"]
+    ELSE
+        FOR    ${index}    ${workbench_title}    IN ENUMERATE    @{connected_workbench}
+            Log    ${index}: ${workbench_title}
+            Run Keyword And Continue On Failure    Page Should Contain Element    
+            ...    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/h4/div[text()="${name}"]]/td[@data-label="Connected workbenches"]/ul/li[text()="${workbench_title}"]
+        END
+    END
+
+Data Connection Should Not Be Listed
+    [Documentation]    Checks a Data Connection is not listed in DS Project details page
+    [Arguments]     ${name}
```

Comment:
the difference is that `Data Connection Should Be Listed` checks all the fields of a data connection. I thought to simply the flow by only checking if a row with the DC name exist. It would be running a line of code instead of >10lines